[FR] overhaul of the weather intent, with detailed report

bors-ltd opened the pull request:
The current one would never give the weather condition, as the state was never "clear" or "sunny", but in French already.

The wind description is using the Beaufort scale.

Heavily inspired by the Hungarian template, kudos to RViktor!

I'd like to push it even further, with sentences like "va-t-il pleuvoir/neiger/geler ?", but it requires the Météo-France integration, which not everyone has. I stick to the built-in Met.no integration. Unless there is a way to ship intents along with the Météo-France integration itself.

A more reasonable next step would be "quelle est la météo pour demain ?".

jlpouffier reviewed (changes requested):
Before I even start reviewing this one more closely ☺️ :

Why no test changes?
I would be more comfortable with test changes for this PR to be honest.

Additional note: The build is failing you have an issue in one of your templates ;)

bors-ltd commented:
My bad, I was so excited I forgot the tests. :blush: 

Working on it as we speak.

jlpouffier commented on responses/fr/HassGetWeather.yaml:
```diff
@@ -3,22 +3,200 @@ responses:
+            {% set wind_scale = 'Tempête' %}
+          {% elif wind_speed <= 117 %}
+            {% set wind_scale = 'Violente tempête' %}
+          {% else %}
```
Can you transform this `else` into a `eilf wind_speed > 117`.
I know it should be the same, but I want to avoid showing `vent d'ouragan` if the wind_speed is incorrectly set to `unavailable` or `-1` for example

jlpouffier commented on responses/fr/HassGetWeather.yaml:
```diff
@@ -3,22 +3,200 @@ responses:
+          {% elif wind_speed <= 117 %}
+            {% set wind_scale = 'Violente tempête' %}
+          {% else %}
+            {% set wind_scale = "vent d'ouragan" %}
```
Capitalize? LIke all the other ones?

jlpouffier commented on responses/fr/HassGetWeather.yaml:
```diff
@@ -3,22 +3,200 @@ responses:
+          {% set wind_scale = 'Vent' %}
+        {% endif %}
+
+        {% set wind_bearing_text = '' %}
+        {% if wind_bearing %}
+          {% if wind_bearing < 45 %}
+            {% set wind_bearing_text = 'nord' %}
+          {% elif wind_bearing < 90 %}
+            {% set wind_bearing_text = 'nord-est' %}
+          {% elif wind_bearing < 135 %}
+            {% set wind_bearing_text = 'est' %}
+          {% elif wind_bearing < 180 %}
+            {% set wind_bearing_text = 'sud-est' %}
+          {% elif wind_bearing < 225 %}
+            {% set wind_bearing_text = 'sud' %}
+          {% elif wind_bearing < 270 %}
+            {% set wind_bearing_text = 'sud-ouest' %}
+          {% elif wind_bearing < 315 %}
+            {% set wind_bearing_text = 'ouest' %}
+          {% elif wind_bearing <= 360 %}
+            {% set wind_bearing_text = 'nord-ouest' %}
+          {% else %}
+            {% set wind_bearing_text = 'inconnue' %}
+          {% endif %}
+        {% endif %}
```
I find the sentence "Dont la direction est Ouest" a bit weird.
IN itself it's not that bad but since you already have all the array to translate it, how about you translate a better-sounding sentence inside the translation array itself and use it below?

Something like
```yaml
{% set wind_bearing_text = '' %}
{% if wind_bearing %}
  {% if wind_bearing < 45 %}
    {% set wind_bearing_text = 'venant du nord' %}
...
  {% endif %}
{% endif %}
...
{{ wind_scale }} de {{ wind_speed | int }} {{ wind_speed_unit }}, {{ wind_bearing_text }}, avec des rafales à {{ wind_gust_speed | int }}
...
```

bors-ltd replied:
You're right, it's a leftover of the Hungarian version through an automated translator.

jlpouffier commented on responses/fr/HassGetWeather.yaml:
```diff
@@ -3,22 +3,200 @@ responses:
+          {% endif %}
+        {% endif %}
+
+        {% if wind_speed_unit == 'km/h' %}
+          {% set wind_speed_unit = 'kilomètres par heure' %}
+        {% endif %}
```
The native wid speed unit on the weather entity can be
`m/s, km/h, mi/h, ft/s or kn`

I would say, let's support them all, especially for our french-speaking colleague leaving elsewhere

RSickenberg replied:
In French, `kn` would give `nd` for `noeuds`, idk what is the best :)

jlpouffier commented on responses/fr/HassGetWeather.yaml:
```diff
@@ -3,22 +3,200 @@ responses:
+      uv: >
+        {% set uv_index = state.attributes.get('uv_index') %}
+
+        {% if uv_index %}
```
Does this particular scale have a name?
I quite like the comment on the Beaufort scale for the wind.

bors-ltd replied:
This is the scale I found on the Météo-France website. I could leave a link as the source, but I have no idea how much of a standard it is.

jlpouffier commented on sentences/fr/weather_HassGetWeather.yaml:
```diff
@@ -3,12 +3,53 @@ intents:
+          domain: weather
+        response: uv
+
+      - sentences:
+          - "(<quelest>|<donnemoi>) la vitesse du vent [<aujourdhui>]"
+          - "(À|A) <quel> vitesse le vent souffle [<aujourdhui>]"
+          - "(<quelest>|<donnemoi>) la météo du vent [<aujourdhui>]"
+          - "(<quelest>|<donnemoi>) le détail du vent [<aujourdhui>]"
+        response: wind
+      - sentences:
+          - "(<quelest>|<donnemoi>) la vitesse du vent [<aujourdhui>] à [<le>]{name}"
+          - "(À|A) <quel> vitesse le vent souffle [<aujourdhui>] à [<le>]{name}"
+          - "(<quelest>|<donnemoi>) la météo du vent [<aujourdhui>] (pour|à) [<le>]{name}"
+          - "(<quelest>|<donnemoi>) le détail du vent (pour|à) [<le>]{name}"
+        requires_context:
+          domain: weather
+        response: wind
```
We have a specific device class under the `sensor` domain called `wind_speed`.
I do not think it's a problem we are removing most (if not all) sentences targeting specific device classes under the `sensor` domain.

@piitaya thoughts?

piitaya replied:
If we only target by name, it shouldn't be a issue right?

jlpouffier commented:
Awesome work ❤️

Review done, some comments added

bors-ltd commented:
I find myself duplicating too much code, and it's getting worse. But it doesn't seem possible to have macros shared by all of the response templates.

bors-ltd commented:
And forgot to ask, is there a way to tell if a response is meant to be read or spoken? I'm influenced by the fact my work is destined to my voice assistant. Nothing wrong about displaying "km/h" on the dashboard.

bors-ltd commented:
The more I test it, the more I doubt my own choices.

Like not trying to describe the wind, but just say "Vent de force 3", and you can ask details about the wind speed.

The same with atmospheric pressure, that may be too much information. Its value is in the trend up or down to predict the weather. I could move it to another intent, to begin with.

bors-ltd commented on sentences/fr/weather_HassGetWeather.yaml:
```diff
@@ -3,12 +3,53 @@ intents:
           domain: weather
+
+      - sentences:
+          - "<donnemoi> (un[e]|le) [bulletin|rapport] météo [détaillé[e]]"
```
Just realised with so many optional words, it collides with "donne-moi la météo". Will split in two sentences.

bors-ltd commented:
Updated with the changes I thought about overnight, like just saying the wind force in the detailed report. Instead I moved the air pressure in a separate intent.

I was thinking one way to reduce clutter was having the wind attribute itself provide the Beaufort value. Is that possible through calling the weather integration itself, or some Jinja filter? It looks like the Beaufort scale is used worldwide to justify its integration into HA itself.

I also allowed myself to use smaller sentences like "bulletin météo" or "météo du vent" which are convenient with the text assistant.

Again, I'm open to suggestions on sentences and responses, I'd like something to please everyone, or as many people as possible.

bors-ltd commented:
Updated and rebased with "il y a du vent aujourd'hui". Also simplified the detailed report to just mention wind gusts if any.

bors-ltd commented:
Heavily simplified Beaufort scale computation, now I found about the conversion API in HA core, and borrowed their formula (and rebased).

jlpouffier commented on sentences/fr/_common.yaml:
```diff
@@ -587,6 +588,8 @@ expansion_rules:
   quel: "quel[le][s]"
   quelest: "<quel> (est|sont)"
+  donnemoi: "(donne[s]|dis)[(-| )moi]"
+  what_is_the_class_of_name: "<quelest> (le |la |l'|les )<class> [(indiqué[e][s]|mesuré[e][s]|renvoyé[e][s]|restant[e][s]|retourné[e][s]|utilisé[e][s]|produit[e][s]|consommé[e][s]|donné[e][s]) ][(par|<dans>|sur)] [<le>]{name} [<dans> [<le>]{area}]"
```
```suggestion

```

jlpouffier replied:
Let's remove this one, it's unused anymore.
And then let's finally merge this pr ;) 

Thx a lot man!

bors-ltd commented:
Thanks, I'll also rebase my branch, I see lots of work have happened in the French section, and hit that "Ready for review" button for a final merge.

I was also thinking about posting on some forum to give people they keys to set up a weather report of their liking.

And I'm still around to maintain and evolve this weather intent.
